pkg/scheduler/framework/runtime/framework.go

```diff
@@ -900,33 +900,33 @@ func (f *frameworkImpl) runPreScorePlugin(ctx context.Context, pl framework.PreS
 	return status
 }
 
-// RunScorePlugins runs the set of configured scoring plugins. It returns a list that
-// stores for each scoring plugin name the corresponding NodeScoreList(s).
+// RunScorePlugins runs the set of configured scoring plugins.
+// It returns a list that stores scores from each plugin and total score for each Node.
 // It also returns *Status, which is set to non-success if any of the plugins returns
 // a non-success status.
-func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) (ps framework.PluginToNodeScores, status *framework.Status) {
+func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) (ns []framework.NodePluginScores, status *framework.Status) {
 	startTime := time.Now()
 	defer func() {
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(score, status.Code().String(), f.profileName).Observe(metrics.SinceInSeconds(startTime))
 	}()
-	pluginToNodeScores := make(framework.PluginToNodeScores, len(f.scorePlugins))
-	for _, pl := range f.scorePlugins {
-		pluginToNodeScores[pl.Name()] = make(framework.NodeScoreList, len(nodes))
-	}
+	allNodePluginScores := make([]framework.NodePluginScores, len(nodes))
+	pluginToNodeScores := make([]framework.NodeScoreList, len(f.scorePlugins))
 	ctx, cancel := context.WithCancel(ctx)
 	errCh := parallelize.NewErrorChannel()
 
-	// Run Score method for each node in parallel.
-	f.Parallelizer().Until(ctx, len(nodes), func(index int) {
-		for _, pl := range f.scorePlugins {
-			nodeName := nodes[index].Name
+	// Run Score method of each score plugin in parallel.
+	f.Parallelizer().Until(ctx, len(f.scorePlugins), func(i int) {
+		pl := f.scorePlugins[i]
+		pluginToNodeScores[i] = make(framework.NodeScoreList, len(nodes))
+		for index, node := range nodes {
```

Comment:
It seems not as efficient as before especially we have large amount of nodes.

Reply:
IIUC about goroutine and its behavior, even if a developer divides their code into smaller pieces and runs them in parallel with goroutine, the maximum number of running goroutine is fixed at `GOMAXPROCS` value and that defaults to the value of a number of logical CPUs usable by the current process.

> GOMAXPROCS sets the maximum number of CPUs that can be executing simultaneously and returns the previous setting. It defaults to the value of runtime.NumCPU. If n < 1, it does not change the current setting. This call will go away when the scheduler improves.
https://pkg.go.dev/runtime#GOMAXPROCS

Running something in parallel on many goroutines does not always result in more efficient processing overall. It may become less efficient because of the overhead of context switches if the number of generated goroutines is larger than `GOMAXPROCS`.